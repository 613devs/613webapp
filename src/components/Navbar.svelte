<script lang="ts">
	import { auth, googleProvider } from '$lib/firebaseConfig';
	import { signInWithPopup } from 'firebase/auth';
	import { userStore } from 'sveltefire';

	const user = userStore(auth);
</script>

<header class="navbar">
	<div class="navbar-start prose">
		<h1>613 Stanford Dr</h1>
	</div>

	<div class="navbar-center">
		<a class="btn" href="/">Home</a>
		<a class="btn" href="/blog" data-sveltekit-preload-data>Blog</a>
		<a class="btn" href="/pool" data-sveltekit-preload-data>Pool</a>
	</div>

	<div class="navbar-end space-x-1">
		{#if $user}
			<a class="btn-circle" href="/profile">
				<img
					src={$user.photoURL || 'https://source.unsplash.com/random'}
					alt={$user.displayName}
					class="h-12 w-12 rounded-full"
				/>
			</a>
			<button on:click={() => auth.signOut()} class="btn">Log Out</button>
		{:else}
			<button on:click={() => signInWithPopup(auth, googleProvider)} class="btn">
				<i class="fa-brands fa-google fa-xl" style="color: #eb4d27;" />
				Sign In with Google
			</button>
		{/if}
	</div>
</header>
