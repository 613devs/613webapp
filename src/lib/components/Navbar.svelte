<script lang="ts">
	import { Doc, SignedIn, SignedOut } from 'sveltefire';
	import { login, logout } from '$lib/utils/auth';
</script>

<header class="navbar">
	<div class="navbar-start flex gap-2">
		<div class="prose">
			<h1>613</h1>
		</div>
		<div class="prose hidden md:block">
			<h1>Stanford Dr</h1>
		</div>
	</div>

	<div class="navbar-center space-x-2">
		<a class="btn" href="/">Home</a>
		<a class="btn" href="/pool" data-sveltekit-preload-data>Pool</a>
	</div>

	<div class="navbar-end space-x-1">
		<SignedIn let:user>
			<Doc ref={`profiles/${user.uid}`} let:data>
				<p class="prose hidden md:block">
					Welcome, <span class="italic font-bold">
						{data.username}
					</span>
				</p>
				<a class="btn-circle" href="/profiles/{user?.uid}">
					<img src={user?.photoURL} alt={user?.displayName} class="h-12 w-12 rounded-full" />
				</a>
				<button on:click={logout} class="btn hidden md:block">Log Out</button>
			</Doc>
		</SignedIn>
		<SignedOut>
			<button on:click={login} class="btn">
				<i class="fa-brands fa-google fa-xl" style="color: #eb4d27;" />
				Sign In with Google
			</button>
		</SignedOut>
	</div>
</header>
