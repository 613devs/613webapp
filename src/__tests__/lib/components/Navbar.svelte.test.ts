import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/svelte';
import { mockState } from '../../vitest-setup';
import Navbar from '$lib/components/Navbar.svelte';

describe('Navbar.svelte', () => {
	it('should render logged in', () => {
		const { container } = render(Navbar);
		expect(container.innerHTML).toMatchSnapshot();
	});

	it('should render logged out', () => {
		mockState.isUserSignedIn = false;
		const { container } = render(Navbar);
		expect(container.innerHTML).toMatchSnapshot();
		mockState.isUserSignedIn = true;
	});
});
