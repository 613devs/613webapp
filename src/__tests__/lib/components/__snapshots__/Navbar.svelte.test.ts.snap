// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar.svelte > should render logged in 1`] = `"<div><header class="navbar"><div class="navbar-start flex gap-2"><div class="prose"><h1>613</h1></div> <div class="prose hidden md:block"><h1>Stanford Dr</h1></div></div> <div class="navbar-center space-x-2"><a class="btn" href="/">Home</a> <a class="btn" href="/pool" data-sveltekit-preload-data="">Pool</a></div> <div class="navbar-end space-x-1"><p class="prose hidden md:block">Welcome, <span class="italic font-bold">test user</span></p> <a class="btn-circle" href="/profiles/test-user-uid"><img src="https://example.com/mock-user-photo.jpg" alt="test user" class="h-12 w-12 rounded-full"></a> <button class="btn hidden md:block">Log Out</button><!--<MockDoc>--><!--<MockSignedIn>--> <!--<MockSignedOut>--></div></header><!--<Navbar>--></div>"`;

exports[`Navbar.svelte > should render logged out 1`] = `
"<div><header class="navbar"><div class="navbar-start flex gap-2"><div class="prose"><h1>613</h1></div> <div class="prose hidden md:block"><h1>Stanford Dr</h1></div></div> <div class="navbar-center space-x-2"><a class="btn" href="/">Home</a> <a class="btn" href="/pool" data-sveltekit-preload-data="">Pool</a></div> <div class="navbar-end space-x-1"><!--<MockSignedIn>--> <button class="btn"><i class="fa-brands fa-google fa-xl" style="color: rgb(235, 77, 39);"></i>
				Sign In with Google</button><!--<MockSignedOut>--></div></header><!--<Navbar>--></div>"
`;
